src/operations/blas3/gemm_no_local_partial_vec.hpp

```diff
@@ -356,48 +362,58 @@ class Gemm<input_t, output_t, DoubleBuffer, NbcA, NbcB, ClSize, tile_type,
       auto B = orig_B;
       auto C = orig_C;
 
-      /* 2D register array used to store the result C*/
-      value_t reg_res[item_rows * item_cols];
-      scaling_c<need_check_boundary, a_packet_size, b_packet_size>(
-          reg_res, C, ldc, dim_m_a_start, dim_n_b_start, boundary_check_c,
-          out_of_range);
-      while (k > 0) {
```

Comment:
What's the reason this was changed from a `while` to a `do-while`? How come it has to happen at least once now?

Reply:
This is a minor optimization because k is guaranteed to be at least 1. It probably doesn't make any difference.